src/makeInfinityAdder.js

```diff
@@ -5,6 +5,31 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let total = 0;
+  const adder = (a) => {
+    total = a
+    if (a === undefined) {
+      return 0;
+    }
+
+    const inner = (b) => {
+      if (b === undefined) {
+        const saveTotal = total;
+        total = 0;
+
+        return saveTotal;
+      }
+      total += b;
+
+      return inner;
+    };
+
+    if (a !== undefined) {
```

Comment:
This condition is redundant since the `inner` function is designed to handle the case when `a` is not `undefined`. You can safely remove this check and always return `inner` after the initial call to `adder`.